File: ios/NitroOta/Core/BundleManager.swift
import Foundation
import NitroModules

/**
 * Returns the stored bundle path from user defaults.
 * This method retrieves the full path to the OTA bundle file directly from user defaults.
 *
 * @param context The application context (not needed in iOS)
 * @return The bundle path if available, null otherwise
 */
func getStoredBundlePath() -> String? {
    let preferences = PreferencesUtils()
    return preferences.getOtaUnzippedPath()
}

File: react-native-nitro-ota/android/src/main/java/com/margelo/nitro/nitroota/core/BundleManager.kt
package com.margelo.nitro.nitroota.core

import android.content.Context
import com.margelo.nitro.nitroota.utils.PreferencesUtils

/**
 * Returns the stored bundle path from shared preferences.
 * This method retrieves the full path to the OTA bundle file directly from preferences.
 *
 * @param context The application context
 * @return The bundle path if available, null otherwise
 */
fun getStoredBundlePath(context: Context): String? {
    val preferences = PreferencesUtils.create(context)
    return preferences.getOtaUnzippedPath()
}
